ReText/window.py

```diff
@@ -519,6 +520,107 @@ def dropEvent(e):
+            if action_type is TabActionTypes.CopyFilePath:
+                self._tabs_ctx_menu.addSeparator()
+
+    def _on_tab_context_menu_requested(self, p):
+
+        clicked_tab_index = self.tabWidget.tabBar().tabAt(p)
+        if -1 == clicked_tab_index:
+            raise RuntimeWarning(f"No tab found at [{p.x()};{p.y()}].")
+
+        clicked_tab = self.tabWidget.widget(clicked_tab_index)
+        if not clicked_tab:
+            raise RuntimeWarning(f"No tab found at index {clicked_tab_index}.")
+
+        self.tabWidget.setCurrentIndex(clicked_tab_index)
+
+        for action in self._tabs_ctx_menu_actions.values():
+            action.target_tab_pos = clicked_tab_index
+
+        total_tabs = self.tabWidget.count()
+
+        can_copy_file = clicked_tab and QFileInfo.exists(clicked_tab.fileName)
+        self._tabs_ctx_menu_actions[TabActionTypes.CopyFileName].setEnabled(can_copy_file)
+        self._tabs_ctx_menu_actions[TabActionTypes.CopyFilePath].setEnabled(can_copy_file)
+
+        self._tabs_ctx_menu_actions[TabActionTypes.CloseAll].setEnabled(total_tabs > 1)
+        self._tabs_ctx_menu_actions[TabActionTypes.CloseToLeft].setEnabled(clicked_tab_index > 0)
+        self._tabs_ctx_menu_actions[TabActionTypes.CloseToRight].setEnabled(clicked_tab_index < total_tabs-1)
+        self._tabs_ctx_menu_actions[TabActionTypes.CloseOther].setEnabled(total_tabs > 1)
+
+        has_unmodified = any(not self.tabWidget.widget(i).editBox.document().isModified()
+                             for i in range(self.tabWidget.count()))
+        self._tabs_ctx_menu_actions[TabActionTypes.CloseUnmodified].setEnabled(has_unmodified and total_tabs > 1)
+
+        chosen_action = self._tabs_ctx_menu.exec(p)
+        self._handle_tab_context_action(chosen_action)
+
+    def _handle_tab_context_action(self, action):
+        if not action or action.text() == TabActionTypes.Unknown.value:
+            return
+
+        action_type = TabActionTypes(action.text())
+        curr_tab = action.target_tab_pos
+
+        if action_type in [TabActionTypes.CopyFileName, TabActionTypes.CopyFilePath]:
+            return self._handle_tab_file_copy_action(curr_tab, action_type)
+
+        total_tabs = self.tabWidget.count()
+        last_tab = total_tabs - 1
+        indices_to_close = []
+
+        if action_type is TabActionTypes.Close:
+            indices_to_close = [curr_tab]
+        elif action_type is TabActionTypes.CloseAll:
+            indices_to_close = list(range(last_tab + 1))
+        elif action_type is TabActionTypes.CloseOther:
+            indices_to_close = [i for i in range(last_tab + 1) if i != curr_tab]
+        elif action_type is TabActionTypes.CloseUnmodified:
+            indices_to_close = [i for i in range(last_tab + 1)
+                                if i != curr_tab and not self.tabWidget.widget(i).editBox.document().isModified()]
```

Comment:
Just a suggestion: you can also use `iterateTabs()` method, maybe that will be a bit shorter.